Tools/PowershellModule/src/Library/New-ARMParameterObjects.ps1

```diff
@@ -0,0 +1,494 @@
+    ## Windows Package Manager Functions [WinGet.RestSource.Functions.zip])
+    $CDBDatabaseName    = "WinGet"
+    $CDBContainerName   = "Manifests"
+    
+    if ($RestSourceAuthentication -eq "MicrosoftEntraId") {
+        $ServerAuthenticationType = "microsoftEntraId"
+        $QueryApiValidationEnabled = $true
+    }
+    else {
+        $ServerAuthenticationType = "none"
+        $MicrosoftEntraIdResource = ""
+        $MicrosoftEntraIdResourceScope = ""
+        $QueryApiValidationEnabled = $false
+    }
+
+    ## The values required for Function ARM Template. But not supported in deployment script.
+    $ManifestCacheEndpoint      = ""
+    $MonitoringTenant           = ""
+    $MonitoringRole             = ""
+    $MonitoringMetricsAccount   = ""
+    $RunFromPackageUrl          = ""
+
+    ## Relative Path from the Working Directory to the Azure ARM Template Files
+    $TemplateAppInsightsPath    = "$TemplateFolderPath\applicationinsights.json"
+    $TemplateKeyVaultPath       = "$TemplateFolderPath\keyvault.json"
+    $TemplateStorageAccountPath = "$TemplateFolderPath\storageaccount.json"
+    $TemplateASPPath            = "$TemplateFolderPath\asp.json"
+    $TemplateCDBAccountPath     = "$TemplateFolderPath\cosmosdb.json"
+    $TemplateCDBPath            = "$TemplateFolderPath\cosmosdb-sql.json"
+    $TemplateCDBContainerPath   = "$TemplateFolderPath\cosmosdb-sql-container.json"
+    $TemplateFunctionPath       = "$TemplateFolderPath\azurefunction.json"
+    $TemplateAppConfigPath      = "$TemplateFolderPath\appconfig.json"
+    $TemplateApiManagementPath  = "$TemplateFolderPath\apimanagement.json"
+
+    $ParameterAppInsightsPath    = "$ParameterFolderPath\applicationinsights.json"
+    $ParameterKeyVaultPath       = "$ParameterFolderPath\keyvault.json"
+    $ParameterStorageAccountPath = "$ParameterFolderPath\storageaccount.json"
+    $ParameterASPPath            = "$ParameterFolderPath\asp.json"
+    $ParameterCDBAccountPath     = "$ParameterFolderPath\cosmosdb.json"
+    $ParameterCDBPath            = "$ParameterFolderPath\cosmosdb-sql.json"
+    $ParameterCDBContainerPath   = "$ParameterFolderPath\cosmosdb-sql-container.json"
+    $ParameterFunctionPath       = "$ParameterFolderPath\azurefunction.json"
+    $ParameterAppConfigPath      = "$ParameterFolderPath\appconfig.json"
+    $ParameterApiManagementPath  = "$ParameterFolderPath\apimanagement.json"
+
+    Write-Verbose -Message "ARM Parameter Resource performance is based on the: $ImplementationPerformance."
+    
+    $PrimaryRegionName   = $(Get-AzLocation).Where({$_.Location -eq $Region}).DisplayName
+    $SecondaryRegion     = Get-PairedAzureRegion -Region $Region
+    $SecondaryRegionName = $(Get-AzLocation).Where({$_.Location -eq $SecondaryRegion}).DisplayName
+    
+    Write-Verbose -Message "Retrieving the Azure Tenant and User Information"
+    $AzContext = Get-AzContext
```

Comment:
Do we need to verify $AzContext is not null before proceeding?

Reply:
This is tested earlier in the script with Test-ConnectionToAzure, so null check here seems redundant